commands/http/websocketio.go

```diff
@@ -0,0 +1,57 @@
+package http
+
+import (
+	"io"
+	"time"
+
+	websocket "github.com/gorilla/websocket"
+)
+
+type WebsocketIO struct {
+	conn *websocket.Conn
+	reader io.Reader
+}
+
+func NewWebsocketIO(conn *websocket.Conn) WebsocketIO {
+	return WebsocketIO{conn, nil}
+}
+
+func (wsio WebsocketIO) Read(buf []byte) (int, error) {
+	for {
+		if wsio.reader == nil {
+			_, reader, err := wsio.conn.NextReader()
+			closeError, ok := err.(*websocket.CloseError)
+			if ok && closeError.Code == 1000 {
+				return 0, io.EOF
+			}
+			if err != nil {
+				return 0, err 
+			}
+			wsio.reader = reader
+		}
+		n, err := wsio.reader.Read(buf)
+		if (err != nil) {
+			wsio.reader = nil
+			if n == 0 && err == io.EOF {
+				continue
+			}
+		}
+		return n, err
+	}
+}
+
+func (wsio WebsocketIO) Write(buf []byte) (int, error) {
+	err := wsio.conn.WriteMessage(websocket.BinaryMessage, buf)
+	if err != nil {
+		return 0, err
+	}
+	return len(buf), nil
+}
+
+func (wsio WebsocketIO) Close() error {
+	_ = wsio.conn.WriteControl(
```

Comment:
i do appreciate the explicit discarding of the error here, but is there any case where we care about this error? maybe just throw it in a debug log if its not nil?